src/encoded/types/user_content.py

```diff
@@ -0,0 +1,161 @@
+        'description': 'Listing of all types of content which may be created by people.',
+    })
+class UserContent(Item):
+
+    base_types = ['UserContent'] + Item.base_types
+    schema = load_schema('encoded:schemas/user_content.json')
+    embedded_list = ["submitted_by.display_title"]
+
+    STATUS_ACL = {
+        'released': ALLOW_CURRENT,
+        'archived': ALLOW_CURRENT,
+        'deleted': DELETED,
+        'draft': ONLY_ADMIN_VIEW,
+        'released to project': ALLOW_VIEWING_GROUP_VIEW,
+        'archived to project': ALLOW_VIEWING_GROUP_VIEW
+    }
+
+    def _update(self, properties, sheets=None):
+        if properties.get('name') is None and self.uuid is not None:
+            properties['name'] = str(self.uuid)
+        super(UserContent, self)._update(properties, sheets)
+
+
+
+@collection(
+    name='static-sections',
+    unique_key='user_content:name',
+    properties={
+        'title': 'Static Sections',
+        'description': 'Static Sections for the Portal',
+    })
+class StaticSection(UserContent):
+    """The Software class that contains the software... used."""
+    item_type = 'static_section'
+    schema = load_schema('encoded:schemas/static_section.json')
+
+    @calculated_property(schema={
+        "title": "Content",
+        "description": "Content for the page",
+        "type": "string"
+    })
+    def content(self, request, body=None, file=None):
+
+        if isinstance(body, str) or isinstance(body, dict) or isinstance(body, list):
+            # Don't need to load in anything. We don't currently support dict/json body (via schema) but could in future.
+            return body
+
+        if isinstance(file, str):
+            if file[0:4] == 'http' and '://' in file[4:8]:  # Remote File
+                return get_remote_file_contents(file)
+            else:                                           # Local File
+                file_path = os.path.abspath(os.path.dirname(os.path.realpath(__file__)) + "/../../.." + file)   # Go to top of repo, append file
+                return get_local_file_contents(file_path)
+
+        return None
+
+    @calculated_property(schema={
+        "title": "File Type",
+        "description": "Type of file used for content",
+        "type": "string"
+    })
+    def filetype(self, request, body=None, file=None, options=None):
+        if options and options.get('filetype') is not None:
+            return options['filetype']
+        if isinstance(body, str):
+            return 'txt'
+        if isinstance(body, dict) or isinstance(body, list):
+            return 'json'
+        if isinstance(file, str):
+            filename_parts = file.split('.')
+            if len(filename_parts) > 1:
+                return filename_parts[len(filename_parts) - 1]
+            else:
+                return 'txt' # Default if no file extension.
+        return None
+
+
+
+
+@collection(
+    name='higlass-view-configs',
+    properties={
```

Comment:
Do you not also need/want a uniquekey here?

Reply:
It appears to inherit user_content:name from UserContent w/o issue. Could explicitly add `unique_key="user_content:name"` to it though to add clarity. Seem if add `unique_key="static_section:name"` some errors start to be thrown on insert or create-mapping.